src/cpp/core/r_util/RActiveSessions.cpp

```diff
@@ -86,92 +86,42 @@ Error ActiveSessions::create(const std::string& project,
    return Success();
 }
 
-namespace {
+namespace
+{
 
 bool compareActivityLevel(boost::shared_ptr<ActiveSession> a,
-                          boost::shared_ptr<ActiveSession> b)
+                        boost::shared_ptr<ActiveSession> b)
 {
    return *a > *b;
 }
 
 } // anonymous namespace
 
-std::vector<boost::shared_ptr<ActiveSession> > ActiveSessions::list(
-                                       const FilePath& userHomePath,
-                                       bool projectSharingEnabled) const
+std::vector<boost::shared_ptr<ActiveSession> > ActiveSessions::list(FilePath userHomePath, bool projectSharingEnabled) const
 {
-   // list to return
-   std::vector<boost::shared_ptr<ActiveSession> > sessions;
-
-   // enumerate children and check for sessions
-   std::vector<FilePath> children;
-   Error error = storagePath_.getChildren(children);
-   if (error)
-   {
-      LOG_ERROR(error);
-      return sessions;
-   }
-   std::string prefix = kSessionDirPrefix;
-   for (const FilePath& child : children)
+   std::vector<std::string> sessionIds = storage_->listSessionIds();
+   std::vector<boost::shared_ptr<ActiveSession>> sessions{};
+   for(std::string id : sessionIds)
```

Comment:
Nit: avoid a copy unless necessary
```suggestion
   for(const std::string& id : sessionIds)
```